hack/deployer/runner/eks.go

```diff
@@ -0,0 +1,186 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runner
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+const (
+	EKSDriverID                 = "eks"
+	EKSConfigFileName           = "deployer-config-eks.yml"
+	DefaultEKSRunConfigTemplate = `id: eks-dev
+overrides:
+  clusterName: %s-dev-cluster
+  vaultInfo:
+    address: %s
+    token: %s
+`
+	EKSVaultPath            = "secret/devops-ci/cloud-on-k8s/ci-aws-k8s-operator"
+	clusterCreationTemplate = `apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+metadata:
+  name: {{.ClusterName}}
+  region: {{.Region}}
+  version: "{{.KubernetesVersion}}"
+nodeGroups:
+  - name: ng-1
+    instanceType: {{.MachineType}}
+    desiredCapacity: {{.NodeCount}}
+    ami: static
+    iam:
+      instanceProfileARN: {{.InstanceProfileARN}}
+      instanceRoleARN: {{.InstanceRoleARN}}
+iam:
+  withOIDC: false
+  serviceRoleARN: {{.ServiceRoleARN}}
+`
+	awsAccessKey       = "AWS_ACCESS_KEY"        // nolint
+	awsSecretAccessKey = "AWS_SECRET_ACCESS_KEY" // nolint
+)
+
+func init() {
+	drivers[EKSDriverID] = &EKSDriverFactory{}
+}
+
+type EKSDriverFactory struct {
+}
+
+func (e EKSDriverFactory) Create(plan Plan) (Driver, error) {
+	return &EKSDriver{
+		plan: plan,
+		ctx: map[string]interface{}{
+			"ClusterName":       plan.ClusterName,
+			"Region":            plan.EKS.Region,
+			"KubernetesVersion": plan.KubernetesVersion,
+			"NodeCount":         plan.EKS.NodeCount,
+			"MachineType":       plan.MachineType,
+			"NodeAMI":           plan.EKS.NodeAMI,
+			"WorkDir":           plan.EKS.WorkDir,
```

Comment:
I am following the pattern established in other provider implementations here, but I was wondering whether it would be better to simply pass the plan struct to the template. It would requite encoding aspects of the struct hierarchy into the templates but it would save us from enumerating all used keys here.

Reply:
I think the current approach has the advantage of adding keyvalues that are not in the plan and providing defaults. We do that for OCP and GKE.